Microsoft.Toolkit.Uwp.UI.Controls.Primitives/ConstrainedPresenter/ConstrainedPresenter.cs

```diff
@@ -0,0 +1,119 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// The <see cref="ConstrainedPresenter"/> is a <see cref="ContentPresenter"/> control which can restrict the
+    /// available size for its content based on a scale factor and/or a specific <see cref="AspectRatio"/>.
+    /// </summary>
+    public class ConstrainedPresenter : ContentPresenter // TODO: Think it should be a Border? But it's sealed; hopefully can change in WinUI 3.
```

Comment:
Why should it be a border? Semantically speaking, I would classify this as [Panel](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.panel?view=winrt-20348).

Reply:
Yeah, trying to think why I didn't think it should be a `Panel` probably if you wanted to add border/corner radius or any other garnish to it. But maybe it is simpler to just use as a panel, as we're only using that API from it. @JustinXinLiu thoughts?

Reply:
I did try switching to Panel quick, but it gets a bit more complicated. I think the main thing about it was wanting to restrict the content to a single child vs. multiple.

Reply:
Though the naïve approach to implementing the Panel version seems to still work as expected and allows this construction without another panel inside...

![image](https://user-images.githubusercontent.com/24302614/125333873-976d6980-e2ff-11eb-9a4c-752be7d54260.png)

Reply:
We discussed and decided it should just be a `FrameworkElement` directly (like `Border` and `Viewbox`) however, we can't inherit directly at present, so we'll be a `ContentPresenter` for now. See https://github.com/microsoft/microsoft-ui-xaml/issues/5530